display_list/display_list_complexity.h

```diff
@@ -0,0 +1,195 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLOW_DISPLAY_LIST_COMPLEXITY_H_
+#define FLUTTER_FLOW_DISPLAY_LIST_COMPLEXITY_H_
+
+#include "flutter/display_list/display_list_dispatcher.h"
+#include "flutter/display_list/types.h"
+
+namespace flutter {
+
+class DisplayListComplexityCalculator : public virtual Dispatcher {
```

Comment:
Maybe define it as an abstract base class with 2 methods (complexity and cached) and then the rest of this code will be the "Default/Simple/Fallback/ComplexityCalculator"?